src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -256,10 +262,25 @@ private static String getSkippedMessage(Element testCase) {
         return message;
     }
 
-    public String getDisplayName() {
+    public String getTransformedTestName() {
         return TestNameTransformer.getTransformedName(testName);
     }
 
+    public String getDisplayName() {
+        // Only prepend the enclosing flow node names if there are any and the run this is in has multiple blocks directly containing
+        // test results.
+        if (!getEnclosingFlowNodeNames().isEmpty()) {
```

Comment:
With some tweaks (changing to `scanner.findFirstMatch(getParents(), Collections.singleton(this), FlowScanningUtils.MATCH_BLOCK_START);`, only populating `enclosingId` for `AtomNode` and `BlockStartNode`), this worked. So...if that's the right thing to do, I'll write some more tests and toss up a PR in `workflow-api`.